### ISSUE TITLE
unsupervised GAN without class as conditional input?

### ISSUE THREAD

**Issue by songyh10** [ajbrock/BigGAN-PyTorch#33]
Hi ajbrock,

Thanks a lot for your sharing this great repo! I'm trying to train the model on CIFAR-10 first, but I found that the setting of the training process on CIFAR-10 is similar to ImageNet, which requires a latent vector z and a class label y as inputs. But in most papers, CIFAR-10 is mainly used for unsupervised GAN, which only takes a latent vector z. Therefore, is there a way to train the model on CIFAR-10 in an unsupervised manner? 

By the way, when I directly run the training script of training on CIFAR-10, the log shows that the FID score sometimes turns to NaN. What's the reason behind it?

Thanks you very much!

**Comment by RuiLiFeng** [comment]
In fact the y to generator is sampled randomly. The real label is used in discriminator, where its embedding is contributed to the outputs of discriminator. So if you want an unconditional GAN, you may simply disable the usage of y in discriminator. I recommand you to keep the randomly sampled y in generator, as it is known that random noise will help sharp the generated images. @songyh10

**Comment by songyh10** [comment]
Thanks for your suggestions! @RuiLiFeng
